x-pack/plugins/ml/public/privilege/check_privilege.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+// @ts-ignore
+import { hasLicenseExpired } from '../license/check_license';
+
+import { Privileges } from './common';
+import { getPrivileges } from './get_privileges';
+
+let privileges: Privileges = {};
+
+// Helper type to allow `new Promise(...)`;
+type PromiseConstructor<T> = new (
+  arg: ((resolve: (arg?: T) => void, reject: (arg?: T) => void) => void)
+) => Promise<T>;
+
+// Privileges Promise Constructor
+type PPC = PromiseConstructor<Privileges>;
+// Privileges Promise
+type PP = Promise<Privileges>;
+
+export function checkGetJobsPrivilege(Promise: PPC, kbnUrl: any): PP {
+  return new Promise((resolve, reject) => {
+    getPrivileges().then((priv: Privileges) => {
```

Comment:
`priv`'s type is specified in this function, but not in the others. I never know which is best. TS knows the return type from `getPrivileges`, but having it explicit here is often more readable.

Reply:
Here it's a leftover from when I tried to get the promise-as-function-argument working with TS. I removed it to be in line with the others in bad8f53.